### PR TITLE
Xiaoya add mlflow model registraion for DINO models

### PR DESCRIPTION
**Pull request by xiaoyachong**
The modified segmentation application is able to register models to `MLflow` after training [DLSIA](https://dlsia.readthedocs.io/en/latest/) models or fine-tuning [LightlyTrain](https://docs.lightly.ai/train/stable/semantic_segmentation.html) models.

However, if a LightlyTrain DINO-based model is trained outside the application or without using the built-in model registration workflow, and only a checkpoint file is available (e.g., `model.pt`), the model can still be registered to `MLflow` using a wrapper. This PR is intended for that purpose.

We follow the [LSE model registration](https://github.com/mlexchange/mlex_latent_explorer/tree/main/live_operator_example) workflow. The main difference is that, instead of directly calling `save_mlflow_wrapper.py`, we build and use a container for registration. The reason is that LightlyTrain requires `torch > 2.4`, which is currently not available on macOS. Therefore, we build a container with a tested PyTorch version (`2.9.1`) and perform the registration inside the container. Please refer to the `README` for details.

> Companion PRs:
> mlex_highres: https://github.com/mlexchange/mlex_highres_segmentation/pull/219
> mlex_lightlytrain: https://github.com/mlexchange/mlex_lightlytrain_segmentation_prototype/pull/1

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -157,6 +160,31 @@ services:
...
+    build:
+      context: ./scripts
+      dockerfile: ../Dockerfile.lightly
+    container_name: lightly
+    environment:
+      - MLFLOW_TRACKING_URI_OUTSIDE=http://mlflow:5000
+      - MLFLOW_TRACKING_USERNAME=${MLFLOW_TRACKING_USERNAME}
+      - MLFLOW_TRACKING_PASSWORD=${MLFLOW_TRACKING_PASSWORD}
+      - TILED_API_KEY=${TILED_SINGLE_USER_API_KEY}
+      - DATA_TILED_URI_IMAGES=http://tiled:8000/api/v1/metadata/recons/42tiff
```
Comment (taxe10) [inline docker-compose.yml]: I think these may be accidentally hardcoded?

File: scripts/run_inference.py
Comment (taxe10) [inline scripts/run_inference.py:1]: Is inference needed for model registration or was it added for testing?

File: scripts/run_inference.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                    "finetuned: finetune.py)."
+                )
+                continue
+            pred_mask = run_inference(model, img_uint8)
+            save_results(img_uint8, gt_mask, pred_mask, img_key, inf_cfg, model_name)
+
+        except Exception as e:
+            logger.error(f"Inference failed for '{model_name}': {e}")
+            import traceback
+            traceback.print_exc()
```
Comment (taxe10) [inline scripts/run_inference.py]: `traceback.print_exc()` bypasses the logging system and writes directly to stderr. Replace with `logger.exception()` which logs the full traceback at ERROR level through the same handler.

**Review by taxe10 — COMMENTED** [review]
Nice! I left some minor comments. Before merging, I wanted to flag that we had discussed archiving this repo in favor of moving the master `docker-compose.yml` to the segmentation app. Is there an existing PR for that migration? If so, it might make more sense to move these changes there directly rather than merging here first.

File: scripts/run_inference.py
Comment (xiaoyachong) [reply scripts/run_inference.py:1]: The inference code is intended for testing purposes only. The LSE script does not include inference code, but I added it here in case we need to run inference on a specific dataset and model without using the segmentation app.

The script folder in this repository is independent of the `mlex_lightlytrain` and `mlex_highres` repositories.

File: docker-compose.yml (hunk 1 above)
Comment (xiaoyachong) [reply docker-compose.yml]: I modified the tiled URLs to load from environment variables. These image and mask URLs are used for testing purposes.

File: scripts/run_inference.py (hunk 2 above)
Comment (xiaoyachong) [reply scripts/run_inference.py]: Sounds good. I just modify it.
